### PR TITLE
feature: Allow args to be used with the json formatter

### LINKED ISSUE #741: JSON formatter does not allow arguments
### Current behavior

Currently, the JSON formatter throws an error when you use arguments:
```
Error: spawn <any formatter with args> ENOENT
  at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
  at onErrorNT (node:internal/child_process:478:16)
  at processTicksAndRejections (node:internal/process/task_queues:83:21)
```
Some examples of a formatter with args include:
* `node some-script.js`
* `some-json-formatter arg1 arg2`
* `node wasm-runner.js some-json-formatter.wasm`

### Desired behavior

I want to be able to pass arguments to the formatter process. This would make the formatter option more flexible. Here's an example of a config that has a formatter with an argument:

```JSONC
"json": {
  // ...
  "formatter": "node cucumber-json-formatter.js"
},
```

In this situation I expect node to be spawned with "cucumber-json-formatter.js" as its first argument

### Versions

* **Cypress version**: `10.1.0`
* **Preprocessor version**: `11.1.0`
* **Node version**: `16.15.1`

### Checklist

- [x] I've read the [FAQ](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/faq.md).
- [x] I've read [Instructions for logging issues](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#instructions-for-logging-issues).
- [x] I'm not using `cypress-cucumber-preprocessor@4.3.1` (package name has changed and it is no longer the most recent version, see [#689](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689)).

### PR DESCRIPTION
**Pull request by mrkjdy**
Closes #741

### CODE REVIEW

**Comment by mrkjdy** [comment]
If you prefer, a regular expression can be used instead of a space in the split

**Comment by badeball** [comment]
Actually, I'd rather that the arguments was an entirely separate array-option, as parsing option strings hugely complex.

**Comment by mrkjdy** [comment]
> Actually, I'd rather that the arguments was an entirely separate array-option, as parsing option strings hugely complex.

Made these changes as suggested

File: lib/preprocessor-configuration.ts [hunk 1]
```diff
@@ -172,6 +185,20 @@ function validateEnvironmentOverrides(
     }
   }
 
+  if (hasOwnProperty(environment, "jsonArgs")) {
+    let { jsonArgs } = environment;
+    if (isString(jsonArgs)) {
+      jsonArgs = JSON.parse(jsonArgs);
```
Comment (badeball) [inline lib/preprocessor-configuration.ts]: I'd rather that we don't attempt to do any parsing of this.

**Comment by badeball** [comment]
Missing here is acceptance tests. Are you able to device some cucumber-js tests that verify arg behavior?

**Comment by mrkjdy** [comment]
> Missing here is acceptance tests. Are you able to device some cucumber-js tests that verify arg behavior?

Added a couple tests and improved the `jsonArgs` parsing error message

Also, these two tests are not passing for me locally:
* `step-definitions.test.ts`:
image

* `Scenario: explicit screenshot` (in `json_report.feature`):
image

**Comment by badeball** [comment]
Perfect verification of behavior, very sublime.

As for the failing tests, I'm guessing there was discrepancy between the checked out TS and the compiled JS. I'm not too happy with the fact that one has to remember to build / watch, but it is how it is for now.

I've modified it ever so slightly, by removing JSON parsing, as Cypress actually does this with args that look like JSON (this can be seen by running `cypress open -e 'foo={"bar":"baz"}'` and inspecting the resolved environment).

I also changed to use `assert.match` and took `.exe` extension into consideration (because I'm also testing on Windows in CI).

**Pull request merged**